syntax = "proto3";

package evrpc;

option cc_generic_services = true;

message Rvalue {
    bool success = 1;
    string errmsg = 2;
}

//message SiteKey {
//    bytes key = 1;
//}
//message SiteVal {
//    bytes value = 1;
//}
message Site {
    bytes key = 1;
    bytes value = 2;
}

message PutRequest{
    repeated Site sites = 1;
}
message PutResult{
    repeated Rvalue rvals  = 1;
//    repeated Site sites = 1;
}
message GetRequest{
    repeated bytes keys = 1;
}
message GetResult{
    Rvalue rvals = 1;
    repeated Site sites = 2;
}


service EkvCmd {
    rpc Put (PutRequest) returns (PutResult);
    rpc Get (GetRequest) returns (GetResult);
}

